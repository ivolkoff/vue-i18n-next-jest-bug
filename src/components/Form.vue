<template>
  <div>{{ t('password_recovery') }}</div>
</template>

<script setup>
import { useI18n } from 'vue-i18n';

const { t } = useI18n();
</script>

<i18n>
{
  "en": {
    "email": {
      "placeholder": "E-mail Address"
    },
    "password": {
      "placeholder": "Password"
    },
    "confirmation": {
      "placeholder": "Google Auth code"
    },
    "password_recovery": "Forgot password?",
    "btn": "Sign in"
  },
  "ru": {
    "email": {
      "placeholder": "Email адрес"
    },
    "password": {
      "placeholder": "Пароль"
    },
    "confirmation": {
      "placeholder": "Google Auth код"
    },
    "password_recovery": "Забыли пароль?",
    "btn": "Войти"
  },
  "uk": {
    "email": {
      "placeholder": "Email адреса"
    },
    "password": {
      "placeholder": "Пароль"
    },
    "confirmation": {
      "placeholder": "Google Auth код"
    },
    "password_recovery": "Забули пароль?",
    "btn": "Увійти"
  }
}
</i18n>
