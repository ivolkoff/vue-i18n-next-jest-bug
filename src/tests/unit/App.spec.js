import "@testing-library/jest-dom";
import { createI18n } from "vue-i18n";
import { config, mount } from '@vue/test-utils';
import Form from './../../components/Form.vue';

export const i18n = createI18n({
	legacy: false,
	fallbackLocale: 'en',
	fallbackWarn: false,
	missingWarn: false,
});

config.global.plugins = [
	i18n,
];

describe("App Internationalization", () => {
	test.each(['en', 'ru', 'uk'])('renders correctly %s', locale => {
		i18n.global.locale.value = locale;
		const wrapper = mount(Form);
		expect(wrapper.element).toMatchSnapshot();
	});
});
