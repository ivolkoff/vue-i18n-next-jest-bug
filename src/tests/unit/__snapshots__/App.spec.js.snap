// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App Internationalization renders correctly en 1`] = `
<div>
  password_recovery
</div>
`;

exports[`App Internationalization renders correctly ru 1`] = `
<div>
  password_recovery
</div>
`;

exports[`App Internationalization renders correctly uk 1`] = `
<div>
  password_recovery
</div>
`;
